### PR TITLE
Refactor SyntaxHighlight component and use BEM on it

### LINKED ISSUE #554: Apply BEM to SyntaxHighlight component
Part of #489

### PR DESCRIPTION
**Pull request by y4izus**
Fix #554

### CODE REVIEW

File: components/ui/SyntaxHighlight.vue [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+      size="small"
       title="Copy to clipboard"
+      class="syntax-highlight__copy-button"
       @click="copyToClipboard"
     >
       Copy
-    </button>
-    <pre v-highlightjs><code ref="code" :class="lang"><slot /></code></pre>
+    </cv-button>
+    <pre v-highlightjs="code"><code :class="`${lang} syntax-highlight__code`" /></pre>
```
Comment (delapuente) [inline components/ui/SyntaxHighlight.vue:11]: The [array syntax](https://vuejs.org/v2/guide/class-and-style.html#Array-Syntax) is more succint here and the preferred way when you need to mix dynamic and non-dynamic classes.

File: pages/aer.vue [hunk 2]
```diff
@@ -121,6 +89,36 @@ export default class extends QiskitElementPage {
       description: 'Counts, Memory, Statevector, Unitary, Snapshots'
     }
   ]
+
+  codeExample: string = `from qiskit import QuantumCircuit, execute, Aer, IBMQ
```
Comment (delapuente) [inline pages/aer.vue:93]: Remove the type annotation, it should be inferred from the template string.

File: components/ui/SyntaxHighlight.vue [hunk 3]
```diff
@@ -18,10 +20,10 @@ import { Component, Prop } from 'vue-property-decorator'
 export default class extends Vue {
   @Prop({ type: String, default: 'python' }) lang
   @Prop(String) label
+  @Prop({ type: String, default: '' }) code
```
Comment (delapuente) [inline components/ui/SyntaxHighlight.vue:23]: Non-blocking: add the missing types for this and other properties. If you don't want to make it optional cause you allow the empty string, use:
```suggestion
  @Prop({ type: String, default: '' }) code!: string
```

File: components/ui/SyntaxHighlight.vue (hunk 1 above)
Comment (y4izus) [reply components/ui/SyntaxHighlight.vue:11]: Per offline conversation, I'm going to use `class` and `:class` as shown [here](https://vuejs.org/v2/guide/class-and-style.html#Object-Syntax)

**Pull request merged**